Update stream endpoint to support a userId param

julianbenegas opened the pull request:
<!-------------------------------------------------------------------------
 | Thanks for send a pull request! 🎉
 | First, please make sure you familiar with the contribution guidelines
 | https://github.com/livepeer/livepeer.com/blob/master/CONTRIBUTING.md
 -------------------------------------------------------------------------->

**What does this pull request do? Explain your changes. (required)**

<!-- A clear and concise description of what this pull request does. -->

- `/api/stream` now supports `userId` query param, that filters streams by that userId.
- User streams table now hits `/api/stream`. The old endpoint remains unchanged for backwards compatibility.

<!-- A detailed description of how you tested your code changes. Include details of your testing environment, and the tests you ran to see how your change affects other areas of the code, etc. -->

**Does this pull request close any open issues?**

<!-- Fixes # -->

- closes #256 
- closes #257
- closes #258

**Screenshots (optional):**

<!-- Drag some screenshots here, if applicable -->

**Checklist:**

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] I have read the **CONTRIBUTING** document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

julianbenegas committed:
update stream endpoint to support a userId param

darkdarkdragon commented on packages/api/src/controllers/stream.js:
```diff
@@ -101,11 +101,19 @@ app.get("/", authMiddleware({ admin: true }), async (req, res) => {
     limit = undefined;
   }
 
+  if (userId && req.user.admin !== true && req.user.id !== userId) {
```
This whole endpoint is accessible only for admins (`authMiddleware({ admin: true })`)
So you should remove `{admin: true}` to make it accessible to regular users.

wohlner replied:
Great! Glad we can easily open up this endpoint to regular users. I'm sure you are already doing this, but I want to make sure that a regular users only has access to their own streams, not everyones. Only admins should see all users' streams.

julianbenegas replied:
Yup, that is handled.

julianbenegas committed:
make the endpoint accessible to non admins

julianbenegas committed:
fix permissions

darkdarkdragon reviewed (changes requested):
[unit test](https://github.com/livepeer/livepeer-com/blob/f2a421f962144babc46e8e944b6a3b50377439c6/packages/api/src/controllers/stream.test.js#L380) failing

julianbenegas committed:
fix unit test

julianbenegas commented on packages/api/src/controllers/stream.test.js:
```diff
@@ -373,8 +373,8 @@ describe("controllers/stream", () => {
 
-    it("should not get all streams for admin user", async () => {
-      client.apiKey = adminApiKey;
+    it("should not get all streams for non admin user", async () => {
```
@darkdarkdragon changed the test to test for `non admin user`.

It's a bit weird, the `authMiddleware({ admin: true })` creates the need for a JWT, but the endpoint as is doesn't require it. Do you think I should update the permissions for it to require a JWT? 🤔

darkdarkdragon replied:
> Do you think I should update the permissions for it to require a JWT?

No, there is no need.

darkdarkdragon reviewed (changes requested):
`all` and `nonLivepeerOnly` arguments should be ignored for non-admin users

julianbenegas committed:
ignore some params when userId is defined

darkdarkdragon committed:
/stream:
 - allow 'all' and 'nonLivepeerOnly` fields only for admins
 - do not return 'user' object for non-admins
